[CI] Autoscaling reference docs Incorrect HTTP method for uri

```
java.lang.AssertionError: Failure at [reference/autoscaling/apis/get-autoscaling-decision:11]: expected [2xx] status code but api [raw[method=GET path=_autoscaling/decision/]] returned [405 Method Not Allowed] [{"error":"Incorrect HTTP method for uri [/_autoscaling/decision/?error_trace=true] and method [GET], allowed: [POST]","status":405}] |  
```

Build stats:
https://gradle-enterprise.elastic.co/s/etiuz357uq37a
https://gradle-enterprise.elastic.co/s/ulirtbaby33yw
https://gradle-enterprise.elastic.co/s/scsxzp5ygfots

cc @jasontedor  (because you've recently created these)

--- a/docs/build.gradle
+++ b/docs/build.gradle
@@ -45,8 +45,11 @@ testClusters.integTest {
   if (singleNode().testDistribution == DEFAULT) {
     setting 'xpack.license.self_generated.type', 'trial'
     setting 'indices.lifecycle.history_index_enabled', 'false'
+    if (BuildParams.isSnapshotBuild() == false) {
+      systemProperty 'es.autoscaling_feature_flag_registered', 'true'
+    }
+    setting 'xpack.autoscaling.enabled', 'true'
     if (BuildParams.isSnapshotBuild()) {
-      setting 'xpack.autoscaling.enabled', 'true'
       setting 'xpack.eql.enabled', 'true'
     }
   }

--- a/x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/Autoscaling.java
+++ b/x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/Autoscaling.java
@@ -32,7 +32,7 @@ import java.util.function.Supplier;
  */
 public class Autoscaling extends Plugin implements ActionPlugin {
 
-    private static boolean AUTOSCALING_FEATURE_FLAG_REGISTERED;
+    private static final boolean AUTOSCALING_FEATURE_FLAG_REGISTERED;
 
     static {
         final String property = System.getProperty("es.autoscaling_feature_flag_registered");

--- a/x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/Autoscaling.java
+++ b/x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/Autoscaling.java
@@ -32,6 +32,24 @@ import java.util.function.Supplier;
  */
 public class Autoscaling extends Plugin implements ActionPlugin {
 
+    private static boolean AUTOSCALING_FEATURE_FLAG_REGISTERED;
+
+    static {
+        final String property = System.getProperty("es.autoscaling_feature_flag_registered");
+        if (Build.CURRENT.isSnapshot() && property != null) {
+            throw new IllegalArgumentException("es.autoscaling_feature_flag_registered is only supported in non-snapshot builds");
+        }
+        if ("true".equals(property)) {
+            AUTOSCALING_FEATURE_FLAG_REGISTERED = true;
+        } else if ("false".equals(property) || property == null) {
+            AUTOSCALING_FEATURE_FLAG_REGISTERED = false;
+        } else {
+            throw new IllegalArgumentException(
+                "expected es.autoscaling_feature_flag_registered to be unset or [true|false] but was [" + property + "]"
+            );
+        }
+    }
+
     public static final Setting<Boolean> AUTOSCALING_ENABLED_SETTING = Setting.boolSetting(
         "xpack.autoscaling.enabled",
         false,
@@ -51,7 +69,7 @@ public class Autoscaling extends Plugin implements ActionPlugin {
      */
     @Override
     public List<Setting<?>> getSettings() {
-        if (isSnapshot()) {
+        if (isSnapshot() || AUTOSCALING_FEATURE_FLAG_REGISTERED) {
             return List.of(AUTOSCALING_ENABLED_SETTING);
         } else {
             return List.of();

